### ISSUE TITLE
installed gulp doesn't work after switching to node 0.12

### ISSUE THREAD

**Issue by albertjan** [projectkudu/kudu#1573]
I understand you want to add it to the machine so *it just works* for more people, but I'd like to be able to remove it or update it. I've worked around it locally by removing the installed `gulp` from the path. But this means I have to change my build script for this one case.

Would it be posible to move `C:\DWASFiles\Sites\#1curit-presentation-cloudant\AppData\npm\` up in the path so the `gulp`from there gets picked up before the installed `gulp`

```js
module.js:338
    throw err;
          ^
Error: Cannot find module 'D:\Program Files (x86)\gulp\3.8.10\node_modules\gulp\node_modules\v8flags\3.28.73.flags.json'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (D:\Program Files (x86)\gulp\3.8.10\node_modules\gulp\bin\gulp.js:25:22)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
```

I'm also running into this error: (this with the locally installed version 3.9.0) 

This seems to be resolved if I do a `node node_modules\gulp\bin\gulp.js --version` first

```
:\home\site\repository\node_modules\gulp\node_modules\semver\semver.js:281
    throw new TypeError('Invalid Version: ' + version);
          ^
TypeError: Invalid Version: undefined
    at new SemVer (D:\home\site\repository\node_modules\gulp\node_modules\semver\semver.js:281:11)
    at SemVer.compare (D:\home\site\repository\node_modules\gulp\node_modules\semver\semver.js:348:13)
    at compare (D:\home\site\repository\node_modules\gulp\node_modules\semver\semver.js:571:31)
    at Function.gt (D:\home\site\repository\node_modules\gulp\node_modules\semver\semver.js:600:10)
    at Liftoff.handleArguments (D:\home\site\repository\node_modules\gulp\bin\gulp.js:99:14)
    at Liftoff.<anonymous> (D:\home\site\repository\node_modules\gulp\node_modules\liftoff\index.js:192:16)
    at module.exports (D:\home\site\repository\node_modules\gulp\node_modules\liftoff\node_modules\flagged-respawn\index.js:17:3)
    at Liftoff.<anonymous> (D:\home\site\repository\node_modules\gulp\node_modules\liftoff\index.js:185:9)
    at D:\home\site\repository\node_modules\gulp\node_modules\liftoff\index.js:159:9
    at D:\home\site\repository\node_modules\gulp\node_modules\v8flags\index.js:79:12
```

**Comment by albertjan** [comment]
Ok so using the locally installed version of gulp is easier then modifying the path. 

`node node_modules\gulp\bin\gulp.js`

**Comment by davidebbo** [comment]
We're planning to upgrade to the latest gulp soon, which hopefully will ease some pain.

**Comment by albertjan** [comment]
Great! Could you also move the appdata npm folder up in the path while you're at it :smile:

**Comment by davidebbo** [comment]
Yep, just made the change (cd69d1477d8f4ee16b3ea0995cf437118fcc1e09). Should get deployed to Azure in the next couple days.

**Comment by albertjan** [comment]
Awsome thanks!

**Comment by davidebbo** [comment]
Ok, gulp 3.9.0 is now available.

**Comment by albertjan** [comment]
awesome thanks
On 11-6-2015 02:55:37, David Ebbo <notifications@github.com> wrote:
Ok, gulp 3.9.0 is now available.

**Comment by u-nikos** [comment]
I'm still getting this error on azure when doing the following within deploy.sh (using gulp 3.9.0):
- Run `npm install gulp --global`
- Run `npm install`
- Run `gulp githooks` within the `package.json` `postinstall` script.

Does anyone know how I can fix this?

**Comment by davidebbo** [comment]
This issue was describing two distinct issues. Can you clarify which error you are getting? And are you using the gulp 3.9 that is now pre-installed, or are you bringing your own? Also, are you able to repro the same issue using the Kudu console? This would help isolating away from the deployment script.

**Comment by u-nikos** [comment]
I don't have access to the website at the moment but I'm getting the 'Invalid Version' error. I was able to run the gulp command from within the deployment script but running the gulp command within the npm postinstall script (triggered from the deployment script) throws the invalid version error. With or without the pre-installed gulp. Any ideas?

**Comment by davidebbo** [comment]
If you can find repro steps via Kudu Console that would be ideal. If not, then please [share a repo](https://github.com/projectkudu/kudu/wiki/Using-a-git-repo-to-report-an-issue) that causes the issue.

**Comment by jsheely** [comment]
I am running into this problem as well. The deploy.cmd runs gulp and it errors. If I give it the path 

```
call node_modules\.bin\gulp.cmd build
```

it still fails. If I go to kudu console and use global gulp it fails the same way. But if I go to Kudu console and run gulp.cmd in that folder it runs as expected. Note that I have to be specific and run gulp.cmd not just gulp

**Comment by jsheely** [comment]
Also worth noting. I tried to run gulp directly from gulp.js and resulted in the same semver error.

```
call "!NODE_EXE!" "node_modules\gulp\bin\gulp.js"  build
```

**Comment by jsheely** [comment]
Added a gulp.cmd -v similar to what the original poster described and now my gulp.cmd build works. What is the deal? It makes no sense to me why this works =\

```
ECHO -- Running Gulp Build
REM If you don't do a version print first then gulp.cmd will semver error.
   call gulp.cmd -v 
   call gulp.cmd build
```

**Comment by DefinitelyCarter** [comment]
I had the same issue here as well.  I had to roll back to an earlier version of node to get the deployment to work.

**Comment by davidebbo** [comment]
Note that as of yesterday, we now have Node 4.0 available. Can you give that a try?

**Comment by DefinitelyCarter** [comment]
Having the same issue with v4.0.0

Command: deploy.cmd
Handling node.js deployment.
KuduSync.NET from: 'D:\home\site\repository' to: 'D:\home\site\wwwroot'
Copying file: 'web.config'
The package.json file does not specify node.js engine version constraints.
The node.js application will run with the default node.js version 4.0.0.
Installing Gulp
D:\local\AppData\npm\gulp -> D:\local\AppData\npm\node_modules\gulp\bin\gulp.js
gulp@3.9.0 D:\local\AppData\npm\node_modules\gulp
├── archy@1.0.0
├── interpret@0.6.5
├── pretty-hrtime@1.0.0
├── deprecated@0.0.1
├── minimist@1.2.0
├── tildify@1.1.0 (os-homedir@1.0.1)
├── v8flags@2.0.10 (user-home@1.1.1)
├── chalk@1.1.1 (escape-string-regexp@1.0.3, supports-color@2.0.0, ansi-styles@2.1.0, has-ansi@2.0.0, strip-ansi@3.0.0)
├── semver@4.3.6
├── orchestrator@0.3.7 (sequencify@0.0.7, stream-consume@0.1.0, end-of-stream@0.1.5)
├── liftoff@2.1.0 (extend@2.0.1, rechoir@0.6.2, flagged-respawn@0.3.1, resolve@1.1.6, findup-sync@0.2.1)
├── gulp-util@3.0.6 (array-differ@1.0.0, array-uniq@1.0.2, beeper@1.1.0, lodash._reevaluate@3.0.0, object-assign@3.0.0, lodash._reinterpolate@3.0.0, lodash._reescape@3.0.0, replace-ext@0.0.1, vinyl@0.5.3, lodash.template@3.6.2, multipipe@0.1.2, through2@2.0.0, dateformat@1.0.11)
└── vinyl-fs@0.3.13 (graceful-fs@3.0.8, strip-bom@1.0.0, defaults@1.0.2, vinyl@0.4.6, mkdirp@0.5.1, through2@0.6.5, glob-stream@3.1.18, glob-watcher@0.0.6)
"Executing gulp build"
D:\local\AppData\npm\node_modules\gulp\node_modules\semver\semver.js:281
    throw new TypeError('Invalid Version: ' + version);
Failed exitCode=1, command="D:\local\AppData\npm\gulp.cmd" build
    ^

**Comment by davidebbo** [comment]
Can you please [share a repo](https://github.com/projectkudu/kudu/wiki/Using-a-git-repo-to-report-an-issue) that has the issue? Thanks.

**Comment by jsheely** [comment]
Here is a repo that causes the error. Note the gulp.cmd -v in deploy.cmd. Toggling that will fix/reproduce the error.

It's worth noting that the error did not present itself for the first 3-4 deployments. I changed a few minor things and pushed and the error started appeared every time without the gulp-v

https://github.com/jsheely/azure-gulp-semver-error

**Comment by davidebbo** [comment]
I just tried and was able to deploy it using https://deploy.azure.com/?repository=https://github.com/jsheely/azure-gulp-semver-error. It's strange that the error is not consistent. Maybe if I push a few no-op changes it'll repro?

**Comment by davidebbo** [comment]
Ok, take that back, I had the older Node version. After switching to 4.0 and redeploying, I get that error. Next step will be to isolate it to a simple Kudu Console repro that we can run outside of a deployment.

**Comment by jsheely** [comment]
That's something I've always wondered. How you actually test a deploy.cmd outside of azure/kudu since there are a lot of given variables

**Comment by jsheely** [comment]
Glad you were able to reproduce it though. And just to be clear for everyone. This isn't a node version issue. It's a deployment issue.

It screams of a race condition due to the fact that simply calling gulp -v fixes it. Unless gulp -v configures something in the environment that I'm not aware of.

**Comment by davidebbo** [comment]
I played with it a bunch more, and I don't think it's a race condition. Instead, what I found is that it fails consistently the first time you run gulp on the clean repo, and then it doesn't happen. e.g. try this from Kudu Console:

- Go to `D:\home\site\repository`
- Run `git clean -dxf`
- Run `gulp`. Result: it fails with the *Invalid Version* error.
- Run `gulp` again. Result: it works! You can run it 10 more times and it will always work.
- Run `git clean -dxf` again
- Again, the first `gulp` run will fail, and all others will succeed

What I don't understand is that when I run `git clean -dxf`, it doesn't seem to be deleting any files! Yet clearly it's doing something that's putting it back into the fail-once state.

Also, if you try those same steps with Node 0.10.32, it doesn't happen. i.e. `gulp` works on the first try. But with 0.12.* or 4.0.0, the above steps are 100% consistent.

**Comment by jsheely** [comment]
Interesting because that's like the exact opposite behavior that we are experiencing. Once it stops working it never works again. But if run gulp -v then it will always work for that session.

Your point about Node 0.10.32 was actually my initial thought. That somehow the node path/version was getting assigned late. So when you run gulp it may have been trying to run from version 0.10.32 instead of 0.12.6 (or 4.0.0)

Having said that though I think I disproved that by:
```
console.log('Current node version: ', process.version);
```
in my local gulp file. But I forget honestly. Will try it again

**Comment by davidebbo** [comment]
@jsheely can you try the exact steps in my previous response to check whether you see the same behavior? It would be quite strange if you don't.

**Comment by jsheely** [comment]
I'll try it out.

In other news. I did confirm (again) that it is running 0.12.6 when calling gulp using a local file
```
call node node_modules\gulp\bin\gulp.js
```
Still errors just the same.

**Comment by jsheely** [comment]
Yes I can confirm your steps behave the same way on my machine.

After thinking it through it is having the same behavior as during a deployment. Only difference is gulp -v never calls semver so we get to call gulp without it erroring.

**Comment by jsheely** [comment]
INTERESTING

So running gulp -v after a git clean -dxf I get the just the global. But the second time I get the additional local version

```
PS D:\home\site\repository> gulp -v
[18:09:16] CLI version 3.9.0
PS D:\home\site\repository> gulp -v
[18:09:22] CLI version 3.9.0
[18:09:22] Local version 3.9.0
```

**Comment by jsheely** [comment]
Here is the gulp logic. 

```
function handleArguments(env) {
  if (versionFlag && tasks.length === 0) {
    gutil.log('CLI version', cliPackage.version);
    if (env.modulePackage && typeof env.modulePackage.version !== 'undefined') {
      gutil.log('Local version', env.modulePackage.version);
    }
    process.exit(0);
  }
```
https://github.com/gulpjs/gulp/blob/master/bin/gulp.js#L75

**Comment by davidebbo** [comment]
Oh that's weird. Now the key is to figure out what state the first run is changing to 'fix' the issue. I diff the repo folder in both states and it's identical.

**Comment by davidebbo** [comment]
I found that simply running `git status` is enough to put it back into the 'bad' state. This is very off, since status is supposed to be a readonly command! Yet clearly it's affecting something. On the other hand, `'git diff` doesn't have that effect. So `git clean -dxf` and `git status` are the two *go to bad state* commands I have found.

**Comment by jsheely** [comment]
I believe this bit of code has the issue. Still debugging...

https://github.com/tkellen/js-liftoff/blob/master/index.js#L94

remote: I failed { [Error: ENOTSUP, operation not supported on socket 'D:\home\site\repository\node_modules\gulp\index.j
s']

```
  var modulePath, modulePackage;
  // TODO: break this out into lib/
  // locate local module and package next to config or explicitly provided cwd
  var modulePath, modulePackage;
  try {
    var delim = (process.platform === 'win32' ? ';' : ':'),
        paths = (process.env.NODE_PATH ? process.env.NODE_PATH.split(delim) : []);
    modulePath = resolve.sync(this.moduleName, {basedir: configBase || cwd, paths: paths});
    modulePackage = silentRequire(fileSearch('package.json', [modulePath]));
  } catch (e) {
//jsheely
    console.log('I failed', e);
  }


  // if we have a configuration but we failed to find a local module, maybe
  // we are developing against ourselves?
  if (!modulePath && configPath) {
    // check the package.json sibling to our config to see if its `name`
    // matches the module we're looking for
    var modulePackagePath = fileSearch('package.json', [configBase]);
    modulePackage = silentRequire(modulePackagePath);
    if (modulePackage && modulePackage.name === this.moduleName) {
      // if it does, our module path is `main` inside package.json
      modulePath = path.join(path.dirname(modulePackagePath), modulePackage.main || 'index.js');
      cwd = configBase;
    } else {
      // clear if we just required a package for some other project
      modulePackage = {};
    }
  }
```

**Comment by jsheely** [comment]
Little further down the rabbit hole. I've narrowed the erroring function too
```
glob.sync('D:\home\site\repository\node_modules\gulp\index.js')
```
https://github.com/cowboy/node-findup-sync/blob/master/lib/findup-sync.js#L31

**Comment by jsheely** [comment]
Could be related too. https://github.com/isaacs/node-glob/issues/205

**Comment by davidebbo** [comment]
I think you're on to something, as `d:\home` in indeed a network drive. I RDPed to one of the worker and was still able to repro, but after copying the repo to the local disk, I no longer could.

**Comment by jsheely** [comment]
So I'm thinking that glob is installed as a global package and it's using an old version instead of the updated new version.

**Comment by jsheely** [comment]
Yup. That's it. 

D:\Program Files (x86)\gulp\3.9.0\node_modules\gulp\node_modules\liftoff\node_modules\findup-sync\node_modules\glob

**Comment by davidebbo** [comment]
Hmmm, I tried adding the [fix](https://github.com/isaacs/node-glob/commit/bf3381e90e283624fbd652835e1aefa55d45e2c7) to that file, and it didn't help (I'm RDP'd to one of the workers, so I can make changes). I also tried changing `"D:\Program Files (x86)\gulp\3.9.0\node_modules\gulp\node_modules\liftoff\node_modules\findup-sync\node_modules\glob\glob.js"`, and still no luck. Maybe I didn't do something quite right? Or maybe there are other copies of it somewhere that are affecting things?

**Comment by jsheely** [comment]
sync.js would also need updated in that same folder

**Comment by davidebbo** [comment]
Yes, that did it!

So what should we do here? Looking at https://www.npmjs.com/package/gulp, 3.9.0 is still the latest, and that has the bug. Do we need to wait for gulp to release an update?

**Comment by jsheely** [comment]
Officially that is probably the responsible thing to do even though impatient me says otherwise.

However I'm going back to the beginning now in the gulp.js to figure out why when I run gulp locally it needs to use global gulp (and ultimately the bugged glob).

**Comment by davidebbo** [comment]
Ok, I'll keep this issue active. For now, the workaround of making a dummy call to `gulp -v` is not terribly bad.

Is there even an issue opened on the gulp repo? They don't seem to release updates very often, as it's been 3 months for 3.9.0.

**Comment by jsheely** [comment]
Common node community issue I believe. Modules depend on other modules but don't actively update and follow bugs in the modules they depend on.

**Comment by albertjan** [comment]
I opened up an issue over there when I opened this. Awesome you guys tackled this!

**Comment by jsheely** [comment]
Yea except they closed and locked that one without providing an actual update.

https://github.com/gulpjs/gulp/issues/1088

I'll open a new one and reference this.

**Comment by davidebbo** [comment]
@jsheely @albertjan do you know where this stands? All the referenced issues appear to be closed, and latest gulp is still 3.9.0 :(

**Comment by albertjan** [comment]
Have asked when this will be pushed to npm. https://github.com/gulpjs/gulp/issues/1251#issuecomment-148513697. We have to wait for that to happen, you can install from git but thats not really a __production__ solution.

**Comment by davidebbo** [comment]
Thanks. We are considering patching our Gulp in Azure Web App with the targeted fix above.

**Comment by albertjan** [comment]
@davidebbo I think I may have misunderstood his message gulp doesn't directly require glob. They've update the package that does. So when you install gulp now, you get this:

```sh
albertjan@server:~/test$ npm install gulp
npm WARN engine get-stdin@5.0.0: wanted: {"node":">=0.12.0"} (current: {"node":"0.10.37","npm":"1.4.28"})
gulp@3.9.0 node_modules/gulp
├── interpret@0.6.6
├── pretty-hrtime@1.0.1
├── deprecated@0.0.1
├── archy@1.0.0
├── minimist@1.2.0
├── v8flags@2.0.10 (user-home@1.1.1)
├── tildify@1.1.2 (os-homedir@1.0.1)
├── chalk@1.1.1 (escape-string-regexp@1.0.3, supports-color@2.0.0, ansi-styles@2.1.0, strip-ansi@3.0.0, has-ansi@2.0.0)
├── semver@4.3.6
├── orchestrator@0.3.7 (stream-consume@0.1.0, sequencify@0.0.7, end-of-stream@0.1.5)
├── liftoff@2.2.0 (extend@2.0.1, rechoir@0.6.2, flagged-respawn@0.3.1, findup-sync@0.3.0, resolve@1.1.6)
├── gulp-util@3.0.6 (array-differ@1.0.0, array-uniq@1.0.2, beeper@1.1.0, lodash._reevaluate@3.0.0, lodash._reescape@3.0.0, lodash._reinterpolate@3.0.0, object-assign@3.0.0, replace-0.0.1, lodash.template@3.6.2, vinyl@0.5.3, through2@2.0.0, multipipe@0.1.2, dateformat@1.0.11)
└── vinyl-fs@0.3.14 (graceful-fs@3.0.8, strip-bom@1.0.0, defaults@1.0.3, vinyl@0.4.6, mkdirp@0.5.1, through2@0.6.5, glob-stream@3.1.18, glob-watcher@0.0.6)
albertjan@server:~/test$ npm ls | grep glob
  │ │ └─┬ glob@5.0.15
    ├─┬ glob-stream@3.1.18
    │ ├─┬ glob@4.5.3
    │ ├─┬ glob2base@0.0.12
    ├─┬ glob-watcher@0.0.6
    │   └─┬ globule@0.1.0
    │     ├─┬ glob@3.1.21
```

Which tells us that atleast __some__ of the globs have been updated to 5.x :smile:

**Comment by davidebbo** [comment]
We've patched our gulp with just the targeted fix, so hopefully this issue is gone.

**Comment by xirzec** [comment]
@davidebbo I still see this issue for grunt, any chance you'll patch your local copy of grunt as well? :)

**Comment by davidebbo** [comment]
Yep, we'll look at that. Can you open a separate issue and link to this one?

**Comment by jeremy-holt** [comment]
Using the Kudu command line I installed gulp globally (npm install gulp -g), and my deploy.cmd installs gulp locally. 

When you say you that your "local copy of gulp has been patched" does this mean I should uninstall my global and local versions of gulp?

I continue to get the semver exception with "my" local versions of gulp. (CLI version 3.9.0, Local version 3.9.0)
Node: 4.2.1

I also often see this ....

<img width="811" alt="sshot-1" src="https://cloud.githubusercontent.com/assets/2135027/11086496/6fc8549c-8830-11e5-9048-59113543f419.png">

**Comment by ahmelsayed** [comment]
The window title issue is something to do with PowerShell, and not node or gulp. We saw that before and it was pretty odd and hard to debug. Using cmd rather than powershell for simple commands should work fine. 

The patched version of gulp is the one here `D:\Program Files (x86)\gulp\3.9.0.1\`

**Comment by snobu** [comment]
Check this out:
https://github.com/projectkudu/KuduScript/pull/33

TL;DR: Use `command 2>&1 | out-default`
to avoid this yet to be explained console host behavior.

**Comment by snobu** [comment]
Scratch that.

`command | out-default` is all that's needed.

**Comment by jeremy-holt** [comment]
@ahmelsayed My deploy.cmd looks like this:

```
echo 6. Build gulpfile.js
pushd "%DEPLOYMENT_TARGET%"
IF EXIST "gulpfile.js" ( 
    call .\node_modules\.bin\gulp -v   
    call .\node_modules\.bin\gulp clean
    call .\node_modules\.bin\gulp build    
    call .\node_modules\.bin\gulp bundle
    IF !ERRORLEVEL! NEQ 0 goto error  
)
popd
```
How would I refer to `D:\Program Files (x86)\gulp\3.9.0.1\` - just hard code it?

(I'm afraid my understanding of this script has been limited to 10 days of trial and error copying and pasting other people's scripts!!)

**Comment by ahmelsayed** [comment]
`gulp.cmd` should be [on your path already](https://github.com/projectkudu/kudu/blob/bc3b8917f5d6baaf65af8bb9f630ac6d34bf9cdd/Kudu.Core/Deployment/Generator/ExternalCommandFactory.cs#L108). Just calling `gulp` from a batch file should work.

**Comment by jeremy-holt** [comment]
But should I uninstall the versions of gulp that I manually installed?

**Comment by ahmelsayed** [comment]
you mean the one you installed using `npm install gulp -g`? if you haven't changed the default location for npm global packages using `npm config set prefix <some custom location>`, then `gulp.cmd` would go into `D:\local\AppData\npm` by default. Aside from that location not being on your path by default, it gets wiped out everytime the site restarts. Everything under `D:\local` does. Only stuff under `D:\home` persist. 

try running `where gulp` from cmd or `Get-Command gulp` from powershell and see which one you're using. My guess is you are already using the one under `D:\Program Files (x86)\gulp\3.9.0.1\`

**Comment by jeremy-holt** [comment]
where gulp gives me "D:\Program Files (x86)\gulp\3.9.0.1\".

I ran `npm uninstall gulp -g` and `npm uninstall gulp`.

When I try to run `gulp build` or `"D:\Program Files (x86)\gulp\3.9.0.1\gulp.cmd" build` I get `Local gulp not found in D:\home\site\wwwroot, Try running: npm  install gulp"`.

Running gulp -v gives me CLI version 3.9.0

if I run `npm install gulp` won't this override the one in "D:\Program Files (x86)\gulp\3.9.0.1\" ?

**Comment by jeremy-holt** [comment]
and I haven't changed the default locations of anything

**Comment by ahmelsayed** [comment]
You need gulp installed in both places, globally and locally. The global one acts as a cli which knows how to run gulp tasks while the local one is the one that gets included in your build scripts as a library.

**Comment by jeremy-holt** [comment]
ok - thanks very much for your help

**Comment by jeremy-holt** [comment]
I just wanted to say that everything is working perfectly now.

Also for some reason the deploy time is down from 15 minutes to around 4 minutes.

Many thanks for all of your help.

**Comment by ahmelsayed** [comment]
Awesome! Glad to know everything is working for you now :)

**Comment by Dia6lo** [comment]
I'm getting
`Error: ENOTSUP: operation not supported on socket, scandir 'D:\home\site\wwwroot\src\assets\images\back.png'`
when my deployment script goes to this part:
```
:: 5. Run gulp transformations
IF EXIST "%DEPLOYMENT_TARGET%\gulpfile.js" (
  pushd "%DEPLOYMENT_TARGET%"
  echo Running gulp serve.
  call :ExecuteCmd gulp serve
  IF !ERRORLEVEL! NEQ 0 goto error
  popd
)
```

**Comment by mrmunch** [comment]
I'm getting the same error:
```
remote: [13:20:30] CLI version 3.9.1
remote: [13:20:30] Local version 3.9.1
remote: ........
remote: [13:20:42] Using gulpfile D:\home\site\wwwroot\gulpfile.js
remote: [13:20:42] Starting 'clean'...
remote: [13:20:46] Finished 'clean' after 3.86 s
remote: .........
remote: [13:21:00] Using gulpfile D:\home\site\wwwroot\gulpfile.js
remote: [13:21:00] Starting 'scripts'...
remote: [13:21:04] Starting 'styles'...
remote: [13:21:07] Starting 'stylesAuth'...
remote: [13:21:07] Starting 'styles404'...
remote: [13:21:07] Starting 'copyVendorImages'...
remote: [13:21:07] Starting 'partials'...
remote: [13:21:09] Starting 'fonts'...
remote: ................
remote: events.js:160
remote:       throw er; // Unhandled 'error' event
remote:       ^
remote: An error has occurred during web site deployment.
remote:
remote: Error: ENOTSUP: operation not supported on socket, scandir 'D:\home\site\wwwroot\bower_components\ammap\dist\ammap\images\arrowDown.gif'
remote:     at Error (native)
```

On local it works fine.
Here is a snippet of my deploy.cmd:

```
:: 3. Install npm packages
IF EXIST "%DEPLOYMENT_TARGET%\package.json" (
  pushd "%DEPLOYMENT_TARGET%"
  call :ExecuteCmd !NPM_CMD! install gulp
  call :ExecuteCmd !NPM_CMD! install
  IF !ERRORLEVEL! NEQ 0 goto error
  popd
)

pushd "%DEPLOYMENT_TARGET%"
IF EXIST "gulpfile.js" ( 
    call .\node_modules\.bin\gulp -v   
    call .\node_modules\.bin\gulp clean
    call .\node_modules\.bin\gulp build    
    IF !ERRORLEVEL! NEQ 0 goto error  
)
popd
```
I use node v6.10.0
